scripts/twister

```diff
@@ -206,9 +206,25 @@ sys.path.insert(0, os.path.join(ZEPHYR_BASE, "scripts/pylib/build_helpers"))
 from twisterlib.environment import add_parse_arguments, parse_arguments
 from twisterlib.twister_main import main
 
+
+def python_version_guard():
+    min_ver = (3, 10)
+    if sys.version_info < min_ver:
+        min_ver_str = '.'.join([str(v) for v in min_ver])
+        cur_ver_line = sys.version.replace('\n', '')
+        err_str = "Obsolete Python version!\n" \
```

Comment:
Print out the version that is not supported and use `Unsupported` instead of 'Obsolete`.

example: 'Unsupported python version 3.8"...